### DISCUSSION TITLE
Should you wrap XCTAssertNoDifference.swift inside #if DEBUG ?

### DISCUSSION THREAD

**Discussion by alejandrogarin** [pointfreeco/swift-custom-dump#13] [Q&A] [answered]
Hi guys,

Since XCTAssertNoDifference.swift imports XCTestDynamicOverlay, I guess it should be wrapped inside #if DEBUG ?

Thanks.

**Comment by mbrandonw** [comment, marked as answer]
Hey @alejandrogarin, in the dynamic overlay library we provide an empty implementation of `XCTFail` when not in `DEBUG`:

https://github.com/pointfreeco/xctest-dynamic-overlay/blob/0ec2d710978df8a73ae788b9897d2c189880de0f/Sources/XCTestDynamicOverlay/XCTFail.swift#L74-L93

Are you getting compiler errors when building for release? Can you make sure you are using the newest version by updating your packages?

**Reply by alejandrogarin** [reply]
Thanks for the reply.
I was going to update to TCA 0.25.1 and saw an issue in XCTestDynamicOverlay, I didn't notice the fact that we have an empty implementation ! 

Thanks!

**Comment by mltbnz** [comment]
Kind of related and I'm struggling to find an answer: 
I am using an additional build configuration called `Test` which is a duplicate of `Debug` but I can't use `XCTAssertNoDifference` with it since all test just succeed no matter if there is a difference.
```swift
XCTAssertNoDifference("Blob", "Blob Sr.") -> ✅
```
The diff in `XCTAssertNoDifference()` is created correctly but when I reach the creation of `let failure = ...` its empty 🤷‍♂️

**Reply by davdroman** [reply]
See https://github.com/pointfreeco/swift-composable-architecture/discussions/1519#discussioncomment-3918424
